torch/csrc/jit/codegen/cuda/lower_validation.cpp

```diff
@@ -1037,6 +1037,114 @@ void validateMma(Fusion* fusion) {
   }
 }
 
+namespace {
+
+//! Returns all swizzle ops between the set of iterdomains
+//!  in `from` and `to`.
+std::vector<Expr*> getAllSwizzlesBetween(
+    std::vector<IterDomain*> from,
+    std::vector<IterDomain*> to) {
+  auto all_expr = DependencyCheck::getAllExprsBetween(
+      {from.begin(), from.end()}, {to.begin(), to.end()});
+
+  std::vector<Expr*> all_swizzles;
+
+  std::copy_if(
+      all_expr.begin(),
+      all_expr.end(),
+      std::back_inserter(all_swizzles),
+      [](Expr* expr) {
+        return expr->getExprType().has_value() &&
+            (expr->getExprType().value() == ExprType::Swizzle2D);
+      });
+
+  return all_swizzles;
+}
+
+} // namespace
+
+void validateSwizzle(Fusion* fusion) {
+  auto used_vals = fusion->usedMathVals();
+  for (auto tv : ir_utils::filterByType<TensorView>(used_vals)) {
+    if (tv->hasSwizzleOp()) {
+      // Make sure no swizzle op is inlined:
+      auto inlined_swizzles = getAllSwizzlesBetween(
+          tv->getMaybeRFactorDomain(),
+          {tv->domain()->domain().begin(),
+           tv->domain()->domain().begin() + tv->getComputeAtPosition()});
+      TORCH_INTERNAL_ASSERT(
+          inlined_swizzles.empty(), "No support for inlined swizzles");
+
+      // Make sure thread swizzling only on sharedmem
+      if (tv->getMemoryType() != MemoryType::Shared) {
```

Comment:
Shouldn't it be:
Thread swizzling allowed on global memory and shared memory
Block swizzling allowed on global memory

This would match the logic for the parallelization validation. Maybe you're just waiting until later to build this out, as the implication for these rules is really about when we shift swizzle layouts from expression to expression.

Reply:
Yes currently not yet supporting persistent block swizzling yet. Will build out in follow ups.